includes/derivatives.inc

```diff
@@ -458,7 +458,7 @@ function islandora_large_image_get_args($lossless) {
     $args[] = "-define numrlvls=7";
     $args[] = "-define jp2:tilewidth=1024";
     $args[] = "-define jp2:tileheight=1024";
-    $args[] = "-define jp2:rate=0.02348";
+    $args[] = "-define jp2:quality=42";
```

Comment:
So I am no JP2 master, but according to the Imagemagick docs, the rate parameter relates to compression. 
> Each value is a factor of compression, thus 20 means 20 times compressed.

So our rate `0.02348` means un-compressed?!

Also, by default we specify a single rate but 7 levels. I wonder if are storing multiple uncompressed levels.

For Kakadu, [the default](https://github.com/Islandora/islandora_solution_pack_large_image/blob/7.x/includes/derivatives.inc#L312) is a single layer with a rate of `0.5`. But [normally for large image](https://github.com/Islandora/islandora_solution_pack_large_image/blob/7.x/includes/derivatives.inc#L85) we define 5 layers and pass in 4 rates.

Also, based on the docs for ImageMagick 6.6 it would appear that `numrlvls` should also have the `jp2:` prefix.
http://hiclass.sakura.ne.jp/im/ImageMagick-6.6.7-10/www/jp2.html

So while I think switching this would work, I think also moving the rate to `1` (for a lossless) layer would help and maybe reducing the number of layers to `1` will reduce total filesize as well. 🤷‍♂

Reply:
> So while I think switching this would work, I think also moving the rate to 1 (for a lossless) layer would help and maybe reducing the number of layers to 1 will reduce total filesize as well.

Ok, so if I just add rate back in like this:


Here's the output (compare to the `no_prefix` output in the previous comment):

```
root@dbe9b21e5a14:/var/www/html/testing# exiftool -s -ee -g1 -u -n -D *_prefix_rate1.jp2
---- ExifTool ----
    - ExifToolVersion                 : 10.80
---- System ----
    - FileName                        : numrlvls_7_syntax_test_no_prefix_rate1.jp2
    - Directory                       : .
    - FileSize                        : 1189058
    - FileModifyDate                  : 2019:09:13 17:26:37+00:00
    - FileAccessDate                  : 2019:09:13 17:26:37+00:00
    - FileInodeChangeDate             : 2019:09:13 17:27:03+00:00
    - FilePermissions                 : 640
---- File ----
    - FileType                        : JP2
    - FileTypeExtension               : JP2
    - MIMEType                        : image/jp2
---- Jpeg2000 ----
    0 MajorBrand                      : jp2
    1 MinorVersion                    : 0.0.0
    2 CompatibleBrands                : jp2
    0 ImageHeight                     : 3483
    4 ImageWidth                      : 4833
    8 NumberOfComponents              : 1
   10 BitsPerComponent                : 7
   11 Compression                     : 7
    0 ColorSpecMethod                 : 1
    1 ColorSpecPrecedence             : 0
    2 ColorSpecApproximation          : 0
    3 ColorSpace                      : 17
---- Composite ----
    - ImageSize                       : 4833x3483
    - Megapixels                      : 16.833339
```

Looks identical. I'm still not convinced "rate" does anything (whether that's a bug or a feature I also don't know, but as I said it is referenced in a few of their github issues).

Reply:
> So our rate 0.02348 means un-compressed?!

Here's the full docs, which I hadn't grokked fully:

> `jp2:rate=x,x,...` : the compression ratio values. Each value is a factor of compression, thus 20 means 20 times compressed. The order is from left to right in descending order. A final lossless quality layer is signified by the value 1. The default is a single lossless quality layer.

I got an error when I messed with something today that said the rate (if single) always gets rounded up to 1. So that's why the old code had only a single lossless layer! `0.02348` was interpreted as `1`. And that explains why it looked like it was doing nothing.

Carrying this further, I see I can make a 4x compression, 2x compression, lossless 3-layer JP2 like this:

`convert Z_LARGE_IMAGE_TEST.tiff -define numrlvls=7 -define jp2:tilewidth=1024 -define jp2:rate=4,2,1 -define jp2:tileheight=1024 -define jp2:quality=42 -define jp2:prg=rpcl -define jp2:mode=int -define jp2:prcwidth=16383 -define jp2:prcheight=16383 -define jp2:cblkwidth=64 -define jp2:cblkheight=64 -define jp2:sop numrlvls_7_syntax_test_no_prefix_rate_multiple.jp2`

The filesize is larger, but actually that exif block is otherwise identical, so I don't think that's a good test of what's the same:

```
root@dbe9b21e5a14:/var/www/html/testing# exiftool -s -ee -g1 -u -n -D numrlvls_7_syntax_test_no_prefix_rate_multiple.jp2
---- ExifTool ----
    - ExifToolVersion                 : 10.80
---- System ----
    - FileName                        : numrlvls_7_syntax_test_no_prefix_rate_multiple.jp2
    - Directory                       : .
    - FileSize                        : 7053020
    - FileModifyDate                  : 2019:09:13 17:37:41+00:00
    - FileAccessDate                  : 2019:09:13 17:45:05+00:00
    - FileInodeChangeDate             : 2019:09:13 17:45:58+00:00
    - FilePermissions                 : 640
---- File ----
    - FileType                        : JP2
    - FileTypeExtension               : JP2
    - MIMEType                        : image/jp2
---- Jpeg2000 ----
    0 MajorBrand                      : jp2
    1 MinorVersion                    : 0.0.0
    2 CompatibleBrands                : jp2
    0 ImageHeight                     : 3483
    4 ImageWidth                      : 4833
    8 NumberOfComponents              : 1
   10 BitsPerComponent                : 7
   11 Compression                     : 7
    0 ColorSpecMethod                 : 1
    1 ColorSpecPrecedence             : 0
    2 ColorSpecApproximation          : 0
    3 ColorSpace                      : 17
---- Composite ----
    - ImageSize                       : 4833x3483
    - Megapixels                      : 16.833339
```